File: algorithms/Strings/Medium/ZigZagConversion/VSDZigZagConversion.py

def convert(string,numRowstring):
    if numRows == 1 or numRows >= len(string):
        return string

    L = [''] * numRows
    index, step = 0, 1

    for x in string:
        L[index] += x
        if index == 0:
            step = 1
        elif index == numRows -1:
            step = -1
        index += step

    return ''.join(L)
    


if __name__=="__main__":
    string="PAYPALISHIRING"
    numRows=4
    print(convert(string,numRows))

File: algorithms/Strings/Medium/ZigZagConversion/VSDZigZagConversion.java

public class Main{

    static String convert(String s,int numRows){
        if(numRows == 1 || numRows >= s.length()) return s;

        String[] L = new String[numRows];
        for(int i=0;i<numRows;i++){
            L[i]="";
        }
        int index=0,step=1;
        for(char x:s.toCharArray()){
            L[index]+=x;
            if(index==0) step=1;
            else if(index == numRows-1) step=-1;
            index+=step;
        }
        return String.join("",L);

    }
    public static void main(String[] args) {
        String s="PAYPALISHIRING";
        int numRows=4;
        System.out.println(convert(s,numRows));
    }
}